docs/markdown/Contributions/development/style-guide.md

```diff
@@ -389,6 +389,37 @@ merge_constraints([constraint for constraint in all_constraints if constraint.st
 
 The return type, however, should usually be as precise as possible so that call sites have better type inference.
 
+### Use descriptive names for `TypeVar`s
+
+#### Bounded `TypeVar`s
+
+Bounded `TypeVar`s should be named after their `bound` type, followed by a single letter (starting with `T`, continuing on to `U`, `V`, etc if multiple `TypeVar`s with the same bound are required.):
```

Comment:
Is there an example in our codebase of multiple TypeVars?

Reply:
Those fall into the third category though, so they don't have a suffix. DO we have examples of multiple typevars with the same bound (I'm trying to find out if we need the `U`, `V`, etc... suggestion.

Reply:
OK, to be clear, the point here is for the rule to not be needlessly restrictive should the case arise. The general case is that you should end with `T`, unless there's a good reason why you can't.

Reply:
I'm finding it hard to imagine 2 type vars with the same bound in the same context. I'm not saying it isn't possible, just saying that care is likely super rare.

Reply:
I would rather see `T` always, and then, if required, add a numeric suffix to distinguish. Switching from T (for TypeVar) to other random letters (U and V are not short for anything) is not helpful. Numbers would be better than random letters.